### PR TITLE
Alberta age credit

### LINKED ISSUE #276: Alberta age credit
reference: https://media.icims.com/training/iForms%20Library/Candidate%20Management/Onboarding%20Solution/Alberta%20Personal%20Tax%20Credit.pdf

### PR DESCRIPTION
**Pull request by oliviacaiww**
Fixes #276

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
+  changes:
+    fixed:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:3]:
```suggestion
    added:
```

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]:
```suggestion
- bump: minor
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/base.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  2023-01-01: 5_853
+metadata:
+  unit: currency-CAD
+  label: Alberta age credit base amount 
+  reference:
+    - title: 2023 Alberta Tax Return Form TD1AB
+      href: https://media.icims.com/training/iForms%20Library/Candidate%20Management/Onboarding%20Solution/Alberta%20Personal%20Tax%20Credit.pdf
+    
+
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/base.yaml:12]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/base.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+values:
+  2023-01-01: 5_853
+metadata:
+  unit: currency-CAD
+  label: Alberta age credit base amount 
+  reference:
+    - title: 2023 Alberta Tax Return Form TD1AB
+      href: https://media.icims.com/training/iForms%20Library/Candidate%20Management/Onboarding%20Solution/Alberta%20Personal%20Tax%20Credit.pdf
+    
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/base.yaml:11]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/eligible_age.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+description: Alberta allows people above this age to receive the age credit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/eligible_age.yaml:1]:
```suggestion
description: Alberta provides the age credit to filers above this age threshold.
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/phase_out_rate.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      2023-01-01: 0
+    rate:
+      2023-01-01: 0
+  - threshold:
+      2023-01-01: 43_570
+    rate:
+      2023-01-01: 0.15
+ 
+  
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/phase_out_rate.yaml:23]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/phase_out_rate.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  - threshold:
+      2023-01-01: 0
+    rate:
+      2023-01-01: 0
+  - threshold:
+      2023-01-01: 43_570
+    rate:
+      2023-01-01: 0.15
+ 
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/phase_out_rate.yaml:22]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/phase_out_rate.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+brackets:
+  - threshold:
+      2023-01-01: 0
+    rate:
+      2023-01-01: 0
+  - threshold:
+      2023-01-01: 43_570
+    rate:
+      2023-01-01: 0.15
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/age_credit/phase_out_rate.yaml:21]:
```suggestion
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/age_credit/ab_age_credit_eligible_count.py
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/age_credit/ab_age_credit_eligible_count.py:1]: dont need this as it is all person level

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/age_credit/ab_age_credit.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+class ab_age_credit(Variable):
+    value_type = float
+    entity = Person
+    label = "Alberta age benefit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.AB
+
+    def formula(Person, period, parameters):
+        income = Person("individual_net_income", period)
+        eligibility = Person("ab_age_credit_eligible_count", period)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/age_credit/ab_age_credit.py:13]:
```suggestion
        eligibility = Person("ab_age_credit_eligibility", period)
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/age_credit/ab_age_credit.py [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+        p = parameters(period).gov.provinces.ab.tax.income.credits.age_credit
+        prebenefit = max_((p.base - p.phase_out_rate.calc(income)), 0)
+
+        benefit_amount = select(
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/age_credit/ab_age_credit.py:17]:
```suggestion
        return = where(eligible,  prebenefit * eligibility, 0)
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/age_credit/ab_age_credit_eligibile.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
+from policyengine_canada.model_api import *
+
+
+class ab_age_credit_eligibile(Variable):
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/age_credit/ab_age_credit_eligibile.py:4]:
```suggestion
class ab_age_credit_eligible(Variable):
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**